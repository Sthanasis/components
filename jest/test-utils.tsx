import { render as tlRender, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'src/assets/theme';

const render = (ui: ReactNode, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...tlRender(<ThemeProvider theme={defaultTheme}>{ui}</ThemeProvider>),
  };
};

export { render, fireEvent };
