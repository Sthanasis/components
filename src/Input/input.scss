@import '~/src/css/mixins.scss';

.input {
  @include interactive-element;
  padding-right: 25px;
  &.fullwidth {
    width: -webkit-fill-available;
  }
}
