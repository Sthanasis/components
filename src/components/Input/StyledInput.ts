import { IBaseProps } from 'src/types';
import styled, { CSSObject } from 'styled-components';
import { ThemeType } from 'src/utilities/theme';

interface IInputStaticProps extends IBaseProps {
  theme: ThemeType;
  fullwidth: boolean;
}

export const StyledInput = styled.input(
  ({ fullwidth, theme }: IInputStaticProps) =>
    ({
      padding: theme.padding,
      width: fullwidth ? '-webkit-fill-available' : undefined,
      paddingRight: '25px',
      fontSize: theme.fontSize,
      textDecoration: 'none',
      backgroundColor: 'transparent',
      border: 'none',
      outline: 'none',
      borderRadius: theme.borderRadius,
      display: 'inline-block',
      position: 'relative',
      color: theme.basicPalette.inputText || theme.basicPalette.text,
    } as CSSObject)
);
