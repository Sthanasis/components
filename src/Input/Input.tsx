import { IBaseProps, InputElementType } from '../../../types/props';
import './input.scss';

export interface IInputProps extends IBaseProps {
  type?: InputElementType;
  onChange?: (...args) => any | typeof jest; //eslint-disable-line @typescript-eslint/no-explicit-any
  placeholder?: string;
  fullwidth?: boolean;
  value?: string | number;
  defaultValue?: string | number;
  name?: string;
  required?: boolean;
  onBlur?: (...args) => any | typeof jest; //eslint-disable-line @typescript-eslint/no-explicit-any
  onFocus?: (...args) => any | typeof jest; //eslint-disable-line @typescript-eslint/no-explicit-any
}

const Input = ({
  type = 'text',
  className,
  style = {},
  onChange,
  placeholder,
  fullwidth = false,
  value,
  name,
  required,
  onBlur,
  onFocus,
}: IInputProps): JSX.Element => {
  const classList = ['input'];
  if (className) {
    className.split(' ').forEach((className) => classList.push(className));
  }

  if (fullwidth) classList.push('fullwidth');

  return (
    <input
      name={name}
      type={type}
      style={style}
      className={classList.join(' ')}
      onChange={onChange}
      onBlur={onBlur}
      placeholder={placeholder}
      value={value}
      onFocus={onFocus}
      required={required}
      autoComplete={type === 'password' ? 'on' : 'off'}
    />
  );
};

export default Input;
