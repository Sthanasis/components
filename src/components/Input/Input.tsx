import { IInputProps } from 'src/types';
import { StyledInput } from './StyledInput';

const Input = ({
  type = 'text',
  className,
  style = {},
  onChange,
  placeholder,
  fullwidth = false,
  value,
  name,
  required,
  onBlur,
  onFocus,
}: IInputProps): JSX.Element => (
  <StyledInput
    name={name}
    type={type}
    style={style}
    className={className}
    onChange={onChange}
    fullwidth={fullwidth}
    onBlur={onBlur}
    placeholder={placeholder}
    value={value}
    onFocus={onFocus}
    required={required}
    autoComplete={type === 'password' ? 'on' : 'off'}
  />
);

export default Input;
