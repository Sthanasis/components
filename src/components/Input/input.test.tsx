import { render } from 'test-utils';
import Input from '../Input';

describe('Input Component', () => {
  it('renders without crashing', () => {
    const { getByPlaceholderText } = render(<Input placeholder="test" />);
    expect(getByPlaceholderText('test')).toMatchSnapshot();
  });

  it('passes the classes correctly', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="test" className="test test2" />
    );
    expect(getByPlaceholderText('test')).toHaveClass('test');
  });

  it('takes the fullwidth prop correctly', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="test" fullwidth />
    );
    expect(getByPlaceholderText('test')).toHaveClass('fullwidth');
  });
});
