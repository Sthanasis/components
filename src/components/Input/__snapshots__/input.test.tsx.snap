// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders without crashing 1`] = `
<input
  autocomplete="off"
  class="sc-bcXHqe kQWMzn"
  placeholder="test"
  type="text"
  value=""
/>
`;
